### PR TITLE
[DPS] Per Nicole, add registrationId input parameter for end users to…

### PR DESCRIPTION
**Pull request by helenlivsc**
… play with DPS mini-solution

### CODE REVIEW

File: AZ3166/src/cores/arduino/system/SystemVariables.cpp [hunk 1]
```diff
@@ -24,6 +25,8 @@ int GetMACWithoutColon(char* buff)
         }
     }
 
+    LogInfo("DevKit MAC address: %s", buff);
```
Comment (ihere1) [inline AZ3166/src/cores/arduino/system/SystemVariables.cpp]: Do not call az_iot in system file
 Revert this  file

File: AZ3166/src/cores/arduino/az_iot/provisioning_client/deps/RIoT/Reference/RIoT/Core/RiotCore.c [hunk 2]
```diff
@@ -88,8 +88,46 @@ int RiotStart(uint8_t *CDI, uint16_t CDILen, const char *RegistrationId)
...
-    // Update subject common of alias certificate TBD data to input registration Id
-    x509AliasTBSData.SubjectCommon = RegistrationId;
+    // Verify registrationId input and given an empty input, generate one based on MAC address and firmware version of your DevKit
+    int regIdLength = strlen(RegistrationId);
+    if (regIdLength == 0){
+        char * registrationId = (char*)malloc(AUTO_GEN_REGISTRATION_ID_MAX_LENGTH);
+        memset(registrationId, 0, AUTO_GEN_REGISTRATION_ID_MAX_LENGTH);
+        char * macAddress = GetBoardID();
+        char * fmVersion = getDevkitVersion();
+        LogInfo("DevKit Firmware Version: %s", fmVersion);
```
Comment (ihere1) [inline AZ3166/src/cores/arduino/az_iot/provisioning_client/deps/RIoT/Reference/RIoT/Core/RiotCore.c]: You can LogInfo("%s  %s", fmVersion, macAddress); if you want to log MAC address

File: AZ3166/src/cores/arduino/az_iot/provisioning_client/deps/RIoT/Reference/RIoT/Core/RiotCore.c [hunk 3]
```diff
@@ -88,8 +88,46 @@ int RiotStart(uint8_t *CDI, uint16_t CDILen, const char *RegistrationId)
...
+            if(registrationId[i] == '.'){
+                registrationId[i] = 'v';
+            }
+        }
+
+        // Update subject common of alias certificate TBD data to auto-generated registration Id
+        x509AliasTBSData.SubjectCommon = registrationId;
+    }
+    else if (regIdLength > REGISTRATION_ID_MAX_LENGTH){
+        LogError("Input value for registrationId in DPS.ino exceeds maximum.");
```
Comment (ihere1) [inline AZ3166/src/cores/arduino/az_iot/provisioning_client/deps/RIoT/Reference/RIoT/Core/RiotCore.c]: also print REGISTRATION_ID_MAX_LENGTH here

File: AZ3166/src/cores/arduino/az_iot/provisioning_client/deps/RIoT/Reference/RIoT/Core/RiotCore.c [hunk 4]
```diff
@@ -88,8 +88,46 @@ int RiotStart(uint8_t *CDI, uint16_t CDILen, const char *RegistrationId)
...
+                continue;
+            }
+            else{
+                LogError("Input value for registrationId in DPS.ino does not meet DPS requirements - only alphanumeric, lowercase, and hyphen are supported.");
+                status = RIOT_INVALID_DEVICE_ID;
+                return status;
+            }
+        }
+        // Update subject common of alias certificate TBD data to input registration Id
+        x509AliasTBSData.SubjectCommon = RegistrationId;
```
Comment (ArthurMa1978) [inline AZ3166/src/cores/arduino/az_iot/provisioning_client/deps/RIoT/Reference/RIoT/Core/RiotCore.c]: dup the input RegistrationId

Comment (ArthurMa1978) [reply AZ3166/src/cores/arduino/az_iot/provisioning_client/deps/RIoT/Reference/RIoT/Core/RiotCore.c]: strdup

Comment (ArthurMa1978) [reply AZ3166/src/cores/arduino/az_iot/provisioning_client/deps/RIoT/Reference/RIoT/Core/RiotCore.c]: consider to have free function like RiotFree to free the SubjectCommon.
and in the initialization of x509AliasTBSData this member should be set to NULL.

File: AZ3166/src/cores/arduino/az_iot/provisioning_client/deps/RIoT/Reference/RIoT/Core/RiotCore.c [hunk 5]
```diff
@@ -88,8 +88,46 @@ int RiotStart(uint8_t *CDI, uint16_t CDILen, const char *RegistrationId)
     uint32_t            length;
     uint32_t            dcType;
 
-    // Update subject common of alias certificate TBD data to input registration Id
-    x509AliasTBSData.SubjectCommon = RegistrationId;
+    // Verify registrationId input and given an empty input, generate one based on MAC address and firmware version of your DevKit
```
Comment (ArthurMa1978) [inline AZ3166/src/cores/arduino/az_iot/provisioning_client/deps/RIoT/Reference/RIoT/Core/RiotCore.c]: File name change to DevKitRiotCore, function name too.

File: AZ3166/src/cores/arduino/az_iot/provisioning_client/deps/RIoT/Reference/RIoT/Core/RiotCore.c (hunk 2 above)
Comment (helenlivsc) [reply AZ3166/src/cores/arduino/az_iot/provisioning_client/deps/RIoT/Reference/RIoT/Core/RiotCore.c]: here macAddress is not real MAC address but in the format of az-macaddress, so I get real MAC address from SystemVariables.cpp.

File: AZ3166/src/cores/arduino/az_iot/provisioning_client/deps/RIoT/Reference/RIoT/Core/RiotCore.c (hunk 3 above)
Comment (helenlivsc) [reply AZ3166/src/cores/arduino/az_iot/provisioning_client/deps/RIoT/Reference/RIoT/Core/RiotCore.c]: Actually maximum value has been mentioned in dps.ino. It is fine to add it here again...

File: AZ3166/src/cores/arduino/system/SystemVariables.cpp (hunk 1 above)
Comment (helenlivsc) [reply AZ3166/src/cores/arduino/system/SystemVariables.cpp]: I will use Serial.printf.

File: AZ3166/src/cores/arduino/az_iot/provisioning_client/deps/RIoT/Reference/RIoT/Core/RiotCore.c (hunk 5 above)
Comment (helenlivsc) [reply AZ3166/src/cores/arduino/az_iot/provisioning_client/deps/RIoT/Reference/RIoT/Core/RiotCore.c]: I will refactor all DPS files in a separate PR.

File: AZ3166/src/cores/arduino/az_iot/provisioning_client/deps/RIoT/Reference/RIoT/Core/RiotCore.c (hunk 4 above)
Comment (helenlivsc) [reply AZ3166/src/cores/arduino/az_iot/provisioning_client/deps/RIoT/Reference/RIoT/Core/RiotCore.c]: done.

File: AZ3166/src/cores/arduino/system/SystemVariables.cpp [hunk 6]
```diff
@@ -1,9 +1,11 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. 
+#include "Arduino.h"
 #include "mbed.h"
 #include "EMW10xxInterface.h"
 #include "SystemVariables.h"
 #include "SystemWiFi.h"
+#include "azure_c_shared_utility/xlogging.h"
```
Comment (ihere1) [inline AZ3166/src/cores/arduino/system/SystemVariables.cpp]: Remove this include

File: AZ3166/src/libraries/AzureIoT/examples/DPS/DPS.ino [hunk 7]
```diff
@@ -11,8 +11,12 @@
 #include "utility.h"
 
 // Input DPS instance info
-char* Global_Device_Endpoint = "[Global_Device_Endpoint]";
-char* ID_Scope = "[ID_Scope]";
+char* Global_Device_Endpoint = "global.azure-devices-provisioning.net";
```
Comment (ihere1) [inline AZ3166/src/libraries/AzureIoT/examples/DPS/DPS.ino]: revert

File: AZ3166/src/cores/arduino/system/SystemVariables.cpp [hunk 8]
```diff
@@ -24,6 +25,8 @@ int GetMACWithoutColon(char* buff)
         }
     }
 
+    Serial.printf("DevKit MAC address: %s\r\n", buff);
```
Comment (ArthurMa1978) [inline AZ3166/src/cores/arduino/system/SystemVariables.cpp]: don't print message in such basic function.

**Pull request merged**